node/tests/integration_tests.rs

```diff
@@ -0,0 +1,195 @@
+use rand::prelude::*;
+use simperby_common::{reserved::ReservedState, *};
+use simperby_node::*;
+use simperby_repository::raw::RawRepository;
+
+// TODO: this will be replaced after implementing git server hosting feature.
+const PUBLIC_REPO: &str = "git@github.com:junha1/simperby-temp-repo.git";
+
+fn generate_random_string() -> String {
+    let mut rng = rand::thread_rng();
+    let s1: u128 = rng.gen();
+    let s2: u128 = rng.gen();
+    Hash256::hash(format!("{}{}", s1, s2).as_bytes()).to_string()[0..16].to_owned()
+}
+
+fn gerenate_random_storage_directory() -> String {
+    let temp_dir = std::env::temp_dir();
+    format!(
+        "{}/{}",
+        temp_dir.to_str().unwrap(),
+        generate_random_string()
+    )
+}
+
+fn generate_config(test_name: String, index: usize) -> Config {
+    let (public_key, private_key) = generate_keypair(format!("{}_{}", test_name, index));
+    let storage_dir = gerenate_random_storage_directory();
+
+    Config {
+        chain_name: test_name,
+        public_key,
+        private_key,
+        peer_directory: format!("{}/peers", storage_dir),
+        governance_directory: format!("{}/governance", storage_dir),
+        consensus_directory: format!("{}/consensus", storage_dir),
+        repository_directory: format!("{}/repository", storage_dir),
+        broadcast_interval_ms: None,
+        fetch_interval_ms: None,
+        public_repo_url: vec![PUBLIC_REPO.to_owned()],
+    }
+}
+
+fn get_timestamp() -> Timestamp {
+    std::time::SystemTime::now()
+        .duration_since(std::time::UNIX_EPOCH)
+        .unwrap()
+        .as_millis() as Timestamp
+}
+
+async fn sleep(ms: u64) {
+    tokio::time::sleep(std::time::Duration::from_millis(ms)).await;
+}
+
+async fn run_command(command: &str) {
+    let mut child = tokio::process::Command::new("sh")
+        .arg("-c")
+        .arg(command)
+        .spawn()
+        .expect("failed to execute process");
+    let ecode = child.wait().await.expect("failed to wait on child");
+    assert!(ecode.success());
+}
+
+async fn setup_genesis_repo(path: &str, rs: ReservedState) {
+    run_command(&format!("cd {}", path)).await;
+    run_command("mkdir repo").await;
+    run_command("cd repo && git init").await;
+    run_command("git commit --allow-empty -m 'initial'").await;
+    simperby_repository::raw::reserved_state::write_reserved_state("./", &rs)
+        .await
+        .unwrap();
+    run_command("git add -A").await;
+    run_command("git commit -m 'genesis'").await;
+}
+
+// TODO: remove this
```

Comment:
why is it to be removed?

Reply:
directly accessing the git repository is not needed after we implement a git server